### PR TITLE
Fixing file tabs disappearing when resizing editor pane

### LINKED ISSUE #7016: File tabs disappear when readjusting window size
Repro steps:
1. Have multiple files open (I have 5 files open originally)
2. Adjust the window size of the center panel
Observe that `apstag.js` and `showad.js` disappear from view when the window is too small. They are not present in the drop-down either.

sep-23-2018 10-57-57

### PR DESCRIPTION
**Pull request by Doomwyte**
Fixes #7016

### Summary of Changes

* File tabs now get added/removed from the dropdown when the editor pane is resized

### Videos

nov-17-2018 17-06-16

### CODE REVIEW

File: src/components/Editor/Tabs.js [hunk 1]
```diff
@@ -65,6 +69,10 @@ class Tabs extends PureComponent<Props, State> {
     this.onResize = debounce(() => {
       this.updateHiddenTabs();
     });
+
+    this.onResizeEnd = debounce(() => {
+      this.updateHiddenTabs();``
+    })
```
Comment (bomsy) [inline src/components/Editor/Tabs.js]: This `onResizeEnd` does not seem to be needed.

File: src/components/Editor/Tabs.js [hunk 2]
```diff
@@ -76,10 +84,12 @@ class Tabs extends PureComponent<Props, State> {
   componentDidMount() {
     window.requestIdleCallback(this.updateHiddenTabs);
     window.addEventListener("resize", this.onResize);
+    getEditorContainer(window).addEventListener("resize", this.onResize);
```
Comment (bomsy) [inline src/components/Editor/Tabs.js]: i think its ok to do `document.querySelector(".editor-pane")` directly in here.

File: src/components/Editor/Tabs.js [hunk 3]
```diff
@@ -76,10 +84,12 @@ class Tabs extends PureComponent<Props, State> {
...
   }
 
   componentWillUnmount() {
     window.removeEventListener("resize", this.onResize);
+    getEditorContainer(window).removeEventListener("resize", this.onResize);
```
Comment (bomsy) [inline src/components/Editor/Tabs.js]: nit: and `document.querySelector(".editor-pane")` in here as well

Comment (bomsy) [inline src/components/Editor/Tabs.js]: nit: i think its fine to do this directly in here
```suggestion
    document.querySelector(".editor-pane").removeEventListener("resizeonend", this.onResize);
```

File: src/components/Editor/Tabs.js (hunk 2 above)
Comment (bomsy) [inline src/components/Editor/Tabs.js]: nit: i think its fine to do this directly in here
```suggestion
    getEditorContainer(window).addEventListener("resizeonend", this.onResize);
```

File: src/components/App.js [hunk 4]
```diff
@@ -242,6 +242,13 @@ class App extends Component<Props, State> {
     }));
   }
 
+  triggerEditorPaneResize() {
+    const editorPane = window.document.querySelector(".editor-pane");
+    if (editorPane) {
+      editorPane.dispatchEvent(new Event("resize"));
```
Comment (bomsy) [inline src/components/App.js]: nit: also would it be clearer to dispatch `resizeonend` custom event here
```suggestion
      editorPane.dispatchEvent(new Event("resizeonend"));
```

**Comment by Doomwyte** [comment]
Awesome, thanks! I'll take a look!

**Pull request merged**